### PR TITLE
Initial commit 

### PR DESCRIPTION
**Pull request by tiina303**
Using `OkHttpClient` as that's available for Java8, which seems to be the most popular Java version in use atm & we want folks to be able to integrate even if they are running on older Java versions.

I started from the user API side, i.e. currently sending each request separately. Will work on batching requests next, and will add tests later

### CODE REVIEW

File: posthog/src/test/java/com/posthog/java/AppTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * Unit test for simple App.
+ */
+public class AppTest {
+
+    /**
+     * Would like this to be a separate example file, but couldn't figure out how to
+     * make that work yet
+     */
+    @Test
+    public void example() {
```
Comment (tiina303) [inline posthog/src/test/java/com/posthog/java/AppTest.java:19]: Wanted to make an example file similar to how we have https://github.com/PostHog/posthog-python/blob/master/example.py but I couldn't quite figure out how to get project management tools / dependencies to be my friends.

File: posthog/src/test/java/com/posthog/java/AppTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        String apiKey = "phc_VvgWABT4LDIPbWpNxVKuT92aCAghr0FEFlb2tHnPXBY";
+        // String host = "http://159.65.213.147";
```
Comment (tiina303) [inline posthog/src/test/java/com/posthog/java/AppTest.java:21]: I was just testing against my DigitalOcean test deployment

File: posthog/pom.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tiina303) [inline posthog/pom.xml:1]: haven't looked into this file, so we probably have stuff we don't want here

File: posthog/src/main/java/com/posthog/java/PostHog.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+            return new PostHog(this);
+        }
+    }
+
+    private PostHog(Builder builder) {
+        apiKey = builder.apiKey;
+        host = builder.host;
+    }
+
+    private void send(String distinctId, String event, HashMap<String, Object> properties, String propName) {
```
Comment (yakkomajuri) [inline posthog/src/main/java/com/posthog/java/PostHog.java:51]: let's not have `propName` and `properties` 

`properties`, `$set`, and `$set_once` are not mutually exclusive. I'd add a HashMap for each _or_ you can also put `$set` and `$set_once` _inside_ `properties`. They can be both top-level or nested.

Comment (yakkomajuri) [inline posthog/src/main/java/com/posthog/java/PostHog.java:51]: let's not have `propName` and `properties` 

`properties`, `$set`, and `$set_once` are not mutually exclusive. I'd add a HashMap for each _or_ you can also put `$set` and `$set_once` _inside_ `properties`. They can be both top-level or nested.

File: posthog/src/main/java/com/posthog/java/PostHog.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        send(distinctId, event, properties, "properties");
+    }
+
+    /**
+     * 
+     * @param distinctId which uniquely identifies your user in your database. Must
+     *                   not be null or empty.
+     * @param properties an array with any person properties you'd like to set.
+     */
+    public void identify(String distinctId, HashMap<String, Object> properties) {
```
Comment (yakkomajuri) [inline posthog/src/main/java/com/posthog/java/PostHog.java:90]: Let's accept both `$set` and `$set_once` props here

**Review by yakkomajuri — APPROVED** [review]
overall looks good as a start, a few comments but don't let me be a blocker - happy for you to move fast here

File: posthog/src/main/java/com/posthog/java/PostHog.java (hunk 4 above)
Comment (tiina303) [reply posthog/src/main/java/com/posthog/java/PostHog.java:51]: Thanks, didn't know they can be nested too, that seems better & easier to make clean.

**Pull request merged**